openff/interchange/interop/openmm/_import/compat.py

```diff
@@ -0,0 +1,35 @@
+from typing import Union
+
+import openmm
+import openmm.app
+from openff.toolkit import Topology
+
+from openff.interchange.exceptions import UnsupportedImportError
+
+
+def _check_compatible_inputs(
+    system: openmm.System,
+    topology: Union[openmm.app.Topology, Topology, None],
+):
+    """Check that inputs are compatible and supported."""
+    for index in range(system.getNumParticles()):
+        if system.isVirtualSite(index):
+            raise UnsupportedImportError(
+                "A particle is a virtual site, which is not yet supported.",
+            )
+        if system.getParticleMass(index)._value == 0.0:
+            raise UnsupportedImportError(
+                "A particle has a mass of 0.0. Is this intentional?",
+            )
```

Comment:
I think this is likely to cause false positives if users are setting masses to zero to freeze atoms (which is the way OpenMM recommends). I also don't think it's necessary if we're explicitly declining any system with virtual sites!

Reply:
I didn't know this is how atoms are frozen in OpenMM world ... nowhere else does this code handle frozen atoms but I suppose it's not obvious that it would need to?